src/lib/tls/tls13_pqc/hybrid_public_key.cpp

```diff
@@ -0,0 +1,373 @@
+   //   of the two values is injective.
+   return reduce(m_public_keys, std::vector<uint8_t>(), [](auto pkb, const auto& key) {
+      // Technically, this is not correct! `public_key_bits()` is meant to
+      // return a BER-encoded public key. For (e.g.) Kyber, that contract is
+      // broken: It returns the raw encoding as used in the reference
+      // implementation.
+      //
+      // TODO: Provide something like Public_Key::raw_public_key_bits() to
+      //       reflect that difference. Also: Key agreement keys could return
+      //       their raw public value there.
+      return concat(pkb, key->public_key_bits());
+   });
+}
+
+bool Hybrid_KEM_PublicKey::supports_operation(PublicKeyOperation op) const {
+   return PublicKeyOperation::KeyEncapsulation == op;
+}
+
+namespace {
+
+class Hybrid_KEM_Encryption_Operation final : public PK_Ops::KEM_Encryption_with_KDF {
+   public:
+      Hybrid_KEM_Encryption_Operation(const Hybrid_KEM_PublicKey& key,
+                                      std::string_view kdf,
+                                      std::string_view provider) :
+            PK_Ops::KEM_Encryption_with_KDF(kdf) {
+         std::transform(
+            key.public_keys().begin(),
+            key.public_keys().end(),
+            std::back_inserter(m_kem_encryptors),
+            [&](const auto& pubkey) { return std::make_unique<PK_KEM_Encryptor>(*pubkey, "Raw", provider); });
+      }
+
+      size_t raw_kem_shared_key_length() const override {
+         return reduce(m_kem_encryptors, size_t(0), [](auto acc, const auto& kem_enc) {
+            return acc + kem_enc->shared_key_length(0 /* no KDF */);
+         });
+      }
+
+      size_t encapsulated_key_length() const override {
+         return reduce(m_kem_encryptors, size_t(0), [](auto acc, const auto& kem_enc) {
+            return acc + kem_enc->encapsulated_key_length();
+         });
+      }
+
+      void raw_kem_encrypt(std::span<uint8_t> out_encapsulated_key,
+                           std::span<uint8_t> raw_shared_key,
+                           Botan::RandomNumberGenerator& rng) override {
+         BOTAN_ASSERT_NOMSG(out_encapsulated_key.size() == encapsulated_key_length());
+         BOTAN_ASSERT_NOMSG(raw_shared_key.size() == raw_kem_shared_key_length());
+
+         BufferStuffer encaps_key_stuffer(out_encapsulated_key);
+         BufferStuffer shared_key_stuffer(raw_shared_key);
+
+         for(auto& kem_enc : m_kem_encryptors) {
+            kem_enc->encrypt(encaps_key_stuffer.next(kem_enc->encapsulated_key_length()),
+                             shared_key_stuffer.next(kem_enc->shared_key_length(0 /* no KDF */)),
+                             rng);
+         }
+      }
+
+   private:
+      // Note: PK_KEM_Encryptor can neither be moved nor copied. Hence, we wrap
+      //       it into a std::unique_ptr<> before storing it in a variant/vector.
+      std::vector<std::unique_ptr<PK_KEM_Encryptor>> m_kem_encryptors;
```

Comment:
We should consider making these types movable. (Certainly not copyable in any circumstances given they contain a `unique_ptr` themselves)

 #3641

Reply:
For the record: I gave this one more iteration, removing the `unique_ptr` (now that [`PK_KEM_Decryptor` is movable](https://github.com/randombit/botan/pull/3655)) and simplified the encapsulated key length book-keeping (now that [we have `PK_KEM_Decryptor::encapsulated_key_length()`](https://github.com/randombit/botan/pull/3653))